graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -47,28 +48,48 @@ public TopSecondDegreeByCountForUser(
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
 
-  @Override
-  protected boolean isEdgeEngagementWithinAgeLimit(
+  /**
+   * Return whether the edge is within the age limit which is specified in the request.
+   * It is used to filter information of unwanted edges from being aggregated.
+   * @param request       is the request given by the requester
+   * @param edgeIterator  is the iterator being used to iterate node's edges.
+   *                      It carries information such as the engagement time of the current edge
+   * @return true if this edge is within the max age limit, false otherwise.
+   */
+  private boolean isEdgeEngagementWithinAgeLimit(
       TopSecondDegreeByCountRequestForUser request,
       EdgeIterator edgeIterator) {
     long keepEdgeWithinTime = request.getMaxEdgeEngagementAgeInMillis();
     long edgeEngagementTime = ((TimestampEdgeIterator)edgeIterator).getCurrentEdgeEngagementTimeInMillis();
     return (edgeEngagementTime >= System.currentTimeMillis() - keepEdgeWithinTime);
   }
+  /**
+   * Only social proof types specified in the user request are counted
+   * For example, a request's social proof types only contain "Follow", and a node has "Follow" and "Mention" edges.
+   * Only the "Follow" edge will be counted, and the "Mention" edge is considered invalid
+   */
+  private boolean isEdgeTypeValid(Set<Byte> validEdgeTypes, byte edgeType) {
+    return validEdgeTypes.contains(edgeType);
+  }
 
   @Override
   protected void updateNodeInfo(
+    TopSecondDegreeByCountRequestForUser request,
```

Comment:
it is better not to pass request inside updateNodeInfo function. 

another implementation alternative is to follow the original pattern of isEdgeEngagementWithinAgeLimit. It was outside of updateNodeInfo. how about creating a new, more generic function to replace isEdgeEngagementWithinAgeLimit and moving both isEdgeEngagementWithinAgeLimit and isEdgeTypeValid inside the new function?